cloudflare/resource_cloudflare_record_test.go

```diff
@@ -174,6 +174,31 @@ func TestAccCloudflareRecord_SRV(t *testing.T) {
 	})
 }
 
+func TestAccCloudflareRecord_CAA(t *testing.T) {
+	t.Parallel()
+	var record cloudflare.DNSRecord
+	domain := os.Getenv("CLOUDFLARE_DOMAIN")
+	zoneID := os.Getenv("CLOUDFLARE_ZONE_ID")
+	rnd := generateRandomResourceName()
+	resourceName := fmt.Sprintf("cloudflare_record.%s", rnd)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckCloudflareRecordDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckCloudflareRecordConfigCAA(rnd, zoneID, fmt.Sprintf("tf-acctest-caa.%s", domain)),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudflareRecordExists(resourceName, &record),
+					resource.TestCheckResourceAttr(
```

Comment:
this test doesn't fail if i comment out the changes you've made to `cloudflare/resource_cloudflare_record.go`. is that because we're already saving it to state through the existing map?

Reply:
i suppose the case we want to protect against here is if that map was to go missing in favour of individual assignment so this is expected? 🤔

Reply:
Correct - this was always working but it was never actually guaranteed that it would work. This is only calling out specific values like the lead from other resource types

Reply:
This is as per the discussion in #821